### ISSUE TITLE
reducing thread.sleep in sample leads to crash

### ISSUE THREAD

**Issue by HansHammel** [Eastrall/Ether.Network#73]
reducing the sample clients sleep time like `Thread.Sleep(10);` leads to

> System.ArgumentException: "Offset and length were out of bounds for the array or count is greater than the number of elements from index to the end of the source collection.

in `public static byte[] GetPacketBuffer(byte[] bufferSource, int offset, int size)`

**Comment by Eastrall** [comment]
Thanks for the feedback, I'm going to investigate this issue.

It should be related with the receiving process, something goes wrong with the `DataStartOffset` property.

**Comment by Eastrall** [comment]
I have made some improvements on the receive algorithm for the `NetServer` and `NetClient` on brach : https://github.com/Eastrall/Ether.Network/tree/feature/receiveAlgorithm
I have also reduced the `Thread.Sleep` to 5ms between each send, and it seems to fix the exception. But, at this speed, the fact we call `Console.WriteLine` at each receive process in the handler causes an `StackOverflowException`. This exception is thrown by the `Console.WriteLine` method. Guess we overflow the `BlockingCollection` of the WriteLine method. Shouldn't be a problem in real cases.
